tests/shardddl3/run.sh

```diff
@@ -183,6 +183,77 @@ function DM_082() {
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(1,'aaa');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(2,'bbb');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(3,'ccc');"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} alter index a visible;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(4,'ddd');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(5,'eee');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(6,'fff');"
+    run_sql_source2 "alter table ${shardddl1}.${tb3} alter index a visible;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(7,'ggg');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(8,'hhh');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(9,'iii');"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+function DM_085() {
+    # `ALTER INDEX` not supported in MySQL 5.7, but we setup the second MySQL 8.0 in CI now.
+    run_case 085 "single-source-pessimistic-2" \
+    "run_sql_source2 \"create table ${shardddl1}.${tb1} (a int unique key, b varchar(10));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int unique key, b varchar(10));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb3} (a int unique key, b varchar(10));\"" \
+    "clean_table" "pessimistic"
+    
+    run_case 085 "single-source-optimistic-2" \
+    "run_sql_source2 \"create table ${shardddl1}.${tb1} (a int unique key, b varchar(10));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int unique key, b varchar(10));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb3} (a int unique key, b varchar(10));\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_086_CASE() {
+    run_sql_source2 "alter table ${shardddl1}.${tb1} alter index a visible;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(1,'aaa');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(2,'bbb');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(3,'ccc');"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} alter index a invisible;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(4,'ddd');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(5,'eee');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(6,'fff');"
+    run_sql_source2 "alter table ${shardddl1}.${tb3} alter index a visible;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(7,'ggg');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(8,'hhh');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(9,'iii');"
+    
+    if [[ "$1" = "pessimistic" ]]; then
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "detect inconsistent DDL sequence from source" 1
+    else
+        # schema comparer for optimistic shard DDL can't diff visible/invisible now.
+        # may this needs to be failed?
```

Comment:
seems `show create table` is `  UNIQUE KEY `a` (`a`) /*!80000 INVISIBLE */`, don't know TiDB will use this comment

Reply:
in MySQL 8, it also shows as
```
KEY `c2_idx` (`c2`) /*!80000 INVISIBLE */
```

I think this hint will be used only by the optimizer.